### ISSUE TITLE
Warnings in NativeAOT

### ISSUE THREAD

**Issue by kant2002** [FNA-XNA/FNA#407]
I did try to publish ImGUI.Net application with FNA and notice a lot of warnings. Mostly they are related to usage of `Marshal.SizeOf(typeof(T))`. 
I did try take a look if they can be replaced with calls `Marshal.SizeOf<T>()` which did not produce any warnings.

I think it would be valuable if FNA not only 100% works in NativeAOT, but also did not have any warnigns, since they can hide real issues in non FNA code.
I see two options
1. Bump TFM to 4.51
2. Embed code analysis attributes in assembly and suppress all warnings because they are safe to use.

Personally I prefer option 1, since they are less hacky, but really would like to know your opinion.

**Comment by TheSpydog** [comment]
Agreed, this is a problem we'll need to address. There is [a PR open](https://github.com/FNA-XNA/FNA/pull/403) to fix some of the Marshal.SizeOf calls at least.

Regarding the options you mentioned, .NET 4.0 is and will always be our minimum spec, because FNA is primarily a preservation project. We even have to make sure everything builds with Visual Studio 2010! But where necessary, we can wrap annotations inside `#if NET7_0_OR_GREATER` flags, like that PR's MarshalHelper does.
